### PR TITLE
Adding Treasure TYPE-9 series III

### PR DESCRIPTION
**Pull request by TreasureTypes**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

Adding Treasure TYPE-9 series III

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/treasure/type9s3/info.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            "layout": [
+                { "matrix": [0, 0], "x": 0, "y": 0 },
+                { "matrix": [0, 1], "x": 1, "y": 0 },
+                { "matrix": [0, 2], "x": 2, "y": 0 },
+                { "matrix": [1, 0], "x": 0, "y": 1 },
+                { "matrix": [1, 1], "x": 1, "y": 1 },
+                { "matrix": [1, 2], "x": 2, "y": 1 },
+                { "matrix": [2, 0], "x": 0, "y": 2 },
+                { "matrix": [2, 1], "x": 1, "y": 2 },
+                { "matrix": [2, 2], "x": 2, "y": 2 }
```
Comment (dunk2k) [inline keyboards/treasure/type9s3/info.json:56]: 1. Community layout(s) identified
2. Formatting in `layouts`
```suggestion
    },
    "community_layouts": ["ortho_3x3"],
    "layouts": {
        "LAYOUT_ortho_3x3": {
            "layout": [
                { "matrix": [0, 0], "x": 0, "y": 0 },
                { "matrix": [0, 1], "x": 1, "y": 0 },
                { "matrix": [0, 2], "x": 2, "y": 0 },
                
                { "matrix": [1, 0], "x": 0, "y": 1 },
                { "matrix": [1, 1], "x": 1, "y": 1 },
                { "matrix": [1, 2], "x": 2, "y": 1 },
                
                { "matrix": [2, 0], "x": 0, "y": 2 },
                { "matrix": [2, 1], "x": 1, "y": 2 },
                { "matrix": [2, 2], "x": 2, "y": 2 }
```

File: keyboards/treasure/type9s3/readme.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Treasure TYPE-9 Series III
+
+![TYPE9SIII](https://i.imgur.com/lohJ0Dal.png)
```
Comment (dunk2k) [inline keyboards/treasure/type9s3/readme.md:3]:
```suggestion
![TYPE9SIII](https://i.imgur.com/lohJ0Dalh.png)
```

File: keyboards/treasure/type9s3/readme.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    make treasure/type9s3:default
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+
+Enter the bootloader :
+
+* **Physical reset button**: Briefly press the button on the back of the PCB
```
Comment (dunk2k) [inline keyboards/treasure/type9s3/readme.md:21]: as bootmagic is enabled, can a line regarding using bootmagic to enter bootloader be added?

File: keyboards/treasure/type9s3/info.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        "mousekey": false,
+        "nkro": true,
+        "rgb_matrix": true
+    },
+    "matrix_pins": {
+        "direct": [
+            ["C7", "C6", "D3"],
+            ["B7", "C5", "D2"],
+            ["B5", "B4", "D4"]
+
```
Comment (dunk2k) [inline keyboards/treasure/type9s3/info.json:21]: minor tidy up
```suggestion
```

File: keyboards/treasure/type9s3/config.h
Comment (dunk2k) [inline keyboards/treasure/type9s3/config.h:1]: Can settings in this file that have a data-driven equivalent, be migrated to `info.json` please.

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/treasure/type9s3/readme.md (hunk 2 above)
Comment (TreasureTypes) [reply keyboards/treasure/type9s3/readme.md:3]: what is the difference between lohJ0Dal.png and lohJ0Dalh.png ?

Comment (dunk2k) [reply keyboards/treasure/type9s3/readme.md:3]: > what is the difference between lohJ0Dal.png and lohJ0Dalh.png ?

[quick explanation](https://thomas.vanhoutte.be/miniblog/imgur-thumbnail-trick/)

File: keyboards/treasure/type9s3/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,31 @@
+/* Copyright 2023 Treasure
+*/
```
Comment (waffle87) [inline keyboards/treasure/type9s3/keymaps/default/keymap.c:2]:
```suggestion
// Copyright 2023 Treasure
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/treasure/type9s3/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,31 @@
+/* Copyright 2023 Treasure
+*/
```
Comment (waffle87) [inline keyboards/treasure/type9s3/keymaps/via/keymap.c:2]:
```suggestion
// Copyright 2023 Treasure
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/treasure/type9s3/config.h [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+
+#define RGB_MATRIX_LED_COUNT 9
+#define WS2812_DI_PIN B6
+#define DRIVER_LED_TOTAL 9
+#define RGB_MATRIX_MAXIMUM_BRIGHTNESS 255
```
Comment (zvecr) [inline keyboards/treasure/type9s3/config.h:25]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/treasure/type9s3/config.h
Comment (TreasureTypes) [reply keyboards/treasure/type9s3/config.h:1]: Ok no problem

File: keyboards/treasure/type9s3/readme.md (hunk 3 above)
Comment (TreasureTypes) [reply keyboards/treasure/type9s3/readme.md:21]: Ok i added a line, thanks

File: keyboards/treasure/type9s3/config.h [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+#define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
+#define ENABLE_RGB_MATRIX_CYCLE_UP_DOWN
+#define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
+#define ENABLE_RGB_MATRIX_PIXEL_FRACTAL
+#define ENABLE_RGB_MATRIX_PIXEL_RAIN
+#ifdef RGB_MATRIX_FRAMEBUFFER_EFFECTS
+    #define ENABLE_RGB_MATRIX_TYPING_HEATMAP
+    #define ENABLE_RGB_MATRIX_DIGITAL_RAIN
+#endif
+#ifdef RGB_MATRIX_KEYPRESSES
```
Comment (drashna) [inline keyboards/treasure/type9s3/config.h:28]: These don't need to be conditional.  The effects themselves check for the keypress/keyrelease defines and aren't enabled unless everything needed is defined.

Comment (TreasureTypes) [reply keyboards/treasure/type9s3/config.h]: Ok, i removed the condition

**Review by waffle87 — APPROVED** [review]

**Comment by TreasureTypes** [comment]
Is there something I need to do ?

File: keyboards/treasure/type9s3/info.json [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x5493",
+        "vid": "0x5452"
+    },
+    "ws2812": {
+        "pin": "B6"
+    },
+    "rgb_matrix": {
+        "driver": "WS2812",
```
Comment (drashna) [inline keyboards/treasure/type9s3/info.json:34]:
```suggestion
        "driver": "ws2812",
```

**Review by fauxpark — APPROVED** [review]

File: keyboards/treasure/type9s3/config.h [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+#define ENABLE_RGB_MATRIX_CYCLE_ALL
+#define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
+#define ENABLE_RGB_MATRIX_CYCLE_UP_DOWN
+#define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
+#define ENABLE_RGB_MATRIX_PIXEL_FRACTAL
+#define ENABLE_RGB_MATRIX_PIXEL_RAIN
+#ifdef RGB_MATRIX_FRAMEBUFFER_EFFECTS
+    #define ENABLE_RGB_MATRIX_TYPING_HEATMAP
+    #define ENABLE_RGB_MATRIX_DIGITAL_RAIN
+#endif
```
Comment (drashna) [inline keyboards/treasure/type9s3/config.h:27]: This isn't needed.  The animtion helpers already check this, and if both aren't defined, the animation is not compiled in. 
```suggestion
#define ENABLE_RGB_MATRIX_TYPING_HEATMAP
#define ENABLE_RGB_MATRIX_DIGITAL_RAIN
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/treasure/type9s3/config.h (hunk 10 above)
Comment (TreasureTypes) [reply keyboards/treasure/type9s3/config.h]: ok, noted and changed

**Comment by TreasureTypes** [comment]
Ok I updated the [config.h] file

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thank you!

**Pull request merged**